### PR TITLE
pkg/helm: only inject ownerrefs into namespace-scoped resources

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**
Update helm-operator's custom rendering engine to skip owner reference injection for resources that are cluster-scoped.

**Motivation for the change:**
#1815

### CODE REVIEW

**Comment by hasbro17** [comment]
Add a line to the CHANGELOG if it's a bug fix?

File: pkg/helm/engine/ownerref_test.go [hunk 1]
```diff
@@ -15,8 +15,14 @@
 package engine
 
 import (
+	"fmt"
+	"strings"
 	"testing"
 
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
```
Comment (hasbro17) [inline pkg/helm/engine/ownerref_test.go]: Nit: Remove line between same project imports.

File: pkg/helm/engine/ownerref_test.go [hunk 2]
```diff
@@ -31,53 +37,36 @@ func (e *mockEngine) Render(chrt *chart.Chart, v chartutil.Values) (map[string]s
...
-		"template.yaml":  baseOut,
-		"template2.yaml": baseOut,
-		"empty.yaml":     "",
-		"comment.yaml":   "# This is empty",
+func genTemplate(resourceCount int, withLeadingSep bool, gvk schema.GroupVersionKind, ownerRefs []metav1.OwnerReference) string {
+	sb := &strings.Builder{}
+
+	for i := 0; i < resourceCount; i++ {
+		if withLeadingSep || i > 0 {
+			sb.WriteString("---\n")
```
Comment (hasbro17) [inline pkg/helm/engine/ownerref_test.go]: Nothing to fix, just a passing comment.
https://github.com/golang/go/blob/3a1b4e75f8b6c1b57db73bccf7ca871bf1a97ca9/src/strings/builder.go#L118-L120
```
 // WriteString appends the contents of s to b's buffer.
// It returns the length of s and a nil error.
```

Why return an error in the signature then? I suspect it probably did return an error at one point before they fixed it and decided not break the signature.

Comment (joelanford) [reply pkg/helm/engine/ownerref_test.go]: I bet that it's actually to meet a `StringWriter` interface where other implementations actually do return errors.

File: pkg/helm/release/manager_factory.go [hunk 3]
```diff
@@ -86,13 +86,13 @@ func (f managerFactory) NewManager(cr *unstructured.Unstructured) (Manager, erro
 
 // getReleaseServer creates a ReleaseServer configured with a rendering engine that adds ownerrefs to rendered assets
 // based on the CR.
-func getReleaseServer(cr *unstructured.Unstructured, storageBackend *storage.Storage, tillerKubeClient *kube.Client) (*tiller.ReleaseServer, error) {
+func getReleaseServer(mgr crmanager.Manager, cr *unstructured.Unstructured, storageBackend *storage.Storage, tillerKubeClient *kube.Client) (*tiller.ReleaseServer, error) {
```
Comment (estroz) [inline pkg/helm/release/manager_factory.go]: Will `mgr` be needed for future field access? Perhaps a `meta.RESTMapper` parameter is enough if not.

Comment (joelanford) [reply pkg/helm/release/manager_factory.go]: Thanks for pointing that out. Turns out I can get the `RESTMapper` from `tillerKubeClient`.

**Pull request merged**